packages/libllvm/build.sh

```diff
@@ -88,7 +88,12 @@ termux_step_pre_configure () {
         # see CMakeLists.txt and tools/clang/CMakeLists.txt
 	TERMUX_PKG_EXTRA_CONFIGURE_ARGS+=" -DLLVM_DEFAULT_TARGET_TRIPLE=$LLVM_DEFAULT_TARGET_TRIPLE"
 	TERMUX_PKG_EXTRA_CONFIGURE_ARGS+=" -DLLVM_TARGET_ARCH=$LLVM_TARGET_ARCH -DLLVM_TARGETS_TO_BUILD=all"
-	TERMUX_PKG_EXTRA_CONFIGURE_ARGS+=" -DLLVM_HOST_TRIPLE=$LLVM_DEFAULT_TARGET_TRIPLE"
+	# to get lldb working properly with i686
+        if [ $TERMUX_ARCH = "i686" ]; then
+                TERMUX_PKG_EXTRA_CONFIGURE_ARGS+=" -DLLVM_HOST_TRIPLE=i386-linux-android"
+        else
+                TERMUX_PKG_EXTRA_CONFIGURE_ARGS+=" -DLLVM_HOST_TRIPLE=$LLVM_DEFAULT_TARGET_TRIPLE"
+        fi
```

Comment:
Do we want to change the `LLVM_DEFAULT_TARGET_TRIPLE` variable instead?

That variable is also used in `-DLLVM_DEFAULT_TARGET_TRIPLE=$LLVM_DEFAULT_TARGET_TRIPLE` above, and in the below lines:

```sh
termux_step_post_massage () {
    sed $TERMUX_PKG_BUILDER_DIR/llvm-config.in \
        [...]
        -e "s|@LLVM_DEFAULT_TARGET_TRIPLE@|$LLVM_DEFAULT_TARGET_TRIPLE|g" \
```

Reply:
I don't know for sure but probably not. The process triple (llvm::sys::getProcessTriple();) as far as i can work out should be the most generic triple. The default target should be a default but not neccesarily the lowest. But the whole triple business is weird.